packages/c-table/src/c-table-caption.tsx

```diff
@@ -0,0 +1,41 @@
+import { h, defineComponent, PropType } from "vue"
+import {
+  chakra,
+  ComponentWithProps,
+  DeepPartial,
+  HTMLChakraProps,
+} from "@chakra-ui/vue-system"
+import { useCTableStyles } from "./c-table"
+
+export interface CTableCaptionProps extends HTMLChakraProps<"caption"> {
+  /**
+   * The placement of the table caption. This sets the `caption-side` CSS attribute.
+   * @default "bottom"
+   */
+  placement?: "top" | "bottom"
+}
+
+export const CTableCaption: ComponentWithProps<
+  DeepPartial<CTableCaptionProps>
```

Comment:
With the latest update, we shall be deprecating the `ComponentWithProps<DeepPartial<ComponentType>>` since because of the issues it presents with typing components. Will update you when it's done.

Reply:
Awesome!